bundles/action/org.openhab.action.satel/src/main/java/org/openhab/action/satel/internal/Satel.java

```diff
@@ -0,0 +1,190 @@
+     * <li>event code - Integer}</li>
+     * <li>state restoration flag - Boolean</li>
+     * <li>event description - String</li>
+     * <li>kind of description - Integer</li>
+     * <li>source - Integer</li>
+     * <li>object - Integer</li>
+     * <li>user control number - Integer</li>
+     * <li>next record index - Integer</li>
+     * <li>current record index - Integer</li>
+     * </ol>
+     * 
+     * <p>
+     * The details about each field meaning can be found in protocol documentation.
+     * </p>
+     * 
+     * @param eventIndex event log record index to read
+     * @return event as array of objects
+     */
+    @ActionDoc(text = "Returns record from the event log for given index", returns = "Array of values: timestamp, partition, event class, event code, restore, event description, kind og description, source, object, user control object, next event index, current event index")
+    public static Object[] satelReadEvent(
+            @ParamDoc(name = "eventIndex", text = "index of event to read, -1 for the most recent") int eventIndex) {
+        if (!satelIsConnected()) {
+            return null;
+        }
+        ReadEventCommand cmd = new ReadEventCommand(eventIndex);
+        if (!SatelActionService.satelCommModule.sendCommand(cmd)) {
+            logger.error("Unable to read record for given index: {}", eventIndex);
+            return null;
+        } else if (cmd.isEmpty()) {
+            logger.warn("No record under given index: {}", eventIndex);
+            return null;
+        } else {
+            EventDescription desc = getEventDescription(cmd.getEventCode(), cmd.isRestore());
+            return new Object[] { new DateTimeType(cmd.getTimestamp()), cmd.getPartition(), cmd.getEventClass(),
+                    cmd.getEventCode(), cmd.isRestore(), desc.eventText, desc.descKind, cmd.getSource(),
+                    cmd.getObject(), cmd.getUserControlNumber(), cmd.getNextIndex(), cmd.getCurrentIndex() };
+        }
+    }
+
+    /**
+     * Reads name of alarm system's device.
+     * Device types:
+     * <ul>
+     * <li>PARTITION</li>
+     * <li>ZONE</li>
+     * <li>USER</li>
+     * <li>EXPANDER</li>
+     * <li>LCD</li>
+     * <li>OUTPUT</li>
+     * <li>TIMER</li>
+     * <li>TELEPHONE</li>
+     * <li>OBJECT</li>
+     * </ul>
+     * 
+     * @param deviceType type of the device
+     * @param deviceNumber number of the device
+     * @return string representing device's name or <code>null</code> if device is not present or an error occurred
+     */
+    @ActionDoc(text = "Returns name of a device.")
+    public static String satelReadDeviceName(
+            @ParamDoc(name = "deviceType", text = "type of the device, one of: PARTITION, ZONE, USER, EXPANDER, LCD, OUTPUT, TIMER, TELEPHONE, OBJECT") String deviceType,
+            @ParamDoc(name = "deviceNumber", text = "number of the device to read") int deviceNumber) {
+        if (!satelIsConnected()) {
+            return null;
```

Comment:
Could you please explain what you mean? Failures on sending command are logged. Only warn, but most common reason is invalid device number provided by the user. But if you think "error" would be better here, please let me know.